File: backend/src/main/java/com/NBE_4_5_2/Team5/domain/user/user/service/email/service/DefaultTimeProvider.java
package com.NBE_4_5_2.Team5.domain.user.user.service.email.service;

import org.springframework.stereotype.Component;

import com.NBE_4_5_2.Team5.global.config.email.TimeProvider;

@Component
public class DefaultTimeProvider implements TimeProvider {
	@Override
	public void sleep(long millis) throws InterruptedException {
		Thread.sleep(millis);
	}
}

File: backend/src/main/java/com/NBE_4_5_2/Team5/domain/user/user/service/email/service/DefaultTimeProvider.kt
package com.NBE_4_5_2.Team5.domain.user.user.service.email.service

import com.NBE_4_5_2.Team5.global.config.email.TimeProvider
import org.springframework.stereotype.Component

@Component
class DefaultTimeProvider : TimeProvider {
    @Throws(InterruptedException::class)
    override fun sleep(millis: Long) {
        Thread.sleep(millis)
    }
}
